include/ArchiveEnumerations.h

```diff
@@ -0,0 +1,124 @@
+        FileSystem,      ///< No archive format, raw binary on disk.
+        Zip,             ///< Archive abiding by the PKWARE Zip format.
+        SevenZ           ///< Archive abiding by the 7zip format.
+    };
+
+    /// @brief Used to indicate an algorithm of compression.
+    enum class CompressionMethod
+    {
+        Unknown   = 0,   ///< Uninitialized or Indicates an error condition of some kind.
+        None,            ///< No compression is applied.
+        Deflate,         ///< Common DEFLATE compression.
+        Deflate64,       ///< A 64-bit variant of Deflate.
+        BZip2,           ///< A compression method that is slow to compress, but fast to decompress.
+        ZPAQ,            ///< A compression method that uses a combination of LZ77 and BWT.
+        Brotli,          ///< A google compression method based on LZ77 offering denser compression than DEFLATE.
+        LZMA,            ///< The compression method used by 7zip offering dense compression.
+        LZ4              ///< A compression method focused on compression and decompression speed.
+    };
+
+    /// @brief Used to indicate an algorithm of encryption.
+    enum class EncryptionMethod
+    {
+        Unknown  = 0,    ///< Uninitialized or Indicates an error condition of some kind.
+        None,            ///< No encryption is applied.
+        PKWARE,          ///< Proprietary encryption used by PKWARE in Zip/Zip64 files.
+        AES_128,         ///< A symmetric encryption with a 128-bit key size.
+        AES_192,         ///< A symmetric encryption with a 192-bit key size.
+        AES_256,         ///< A symmetric encryption with a 256-bit key size.
+        RSA_128,         ///< An asymmetric encryption with a 128-bit key size.
+        RSA_192,         ///< An asymmetric encryption with a 192-bit key size.
+        RSA_256,         ///< An asymmetric encryption with a 256-bit key size.
+        TDES_112,        ///< Also 3TES. A symmetric encryption with a 112-bit key size.
+        TDES_168         ///< Also 3TES. A symmetric encryption with a 168-bit key size.
+    };
+
+    /// @brief Used to differentiate entries in some archive systems.
+    enum class EntryType
+    {
+        Unknown   = 0,   ///< Uninitialized or Indicates an error condition of some kind.
+        File,            ///< Indicates the entry is a file.
+        Directory,       ///< Indicates the entry is a directory.
+        Hardlink,        ///< Indicates the entry is a name-file association to data on disk.
+        Symlink          ///< Indicates the entry is a symbolic link (shortcut) to a file or directory entry.
+    };
+
+    /// @brief Used to identify the permissions a file has in an archive.
+    /// @remarks These permissions are based on posix/std::filesystem permissions.
+    enum class FilePermissions : UInt32
+    {
+        None          = 0,           ///< File has no permissions, which is an invalid state.
```

Comment:
Since we a using the unix permission model adding 755 and 644 to this might be a good idea. They are default values for a lot of systems and likely to be need specifically. 755 is Owner has all permissions while group/world have read/execute and 644 is Owner has while group/world have read.

If we care about 750 and 640 might be reasonable entries as well. These are used on more security conscious system for similar reasons sharing reasons as the previous permission set, but other is exclude by default.

Reply:
I have added a couple more helper values as you suggested.  I think I'm otherwise happy with the enum where it is.  Although I'm open to suggestions for another name for "755" and "644" because I'm not thrilled with what I chose, but we can't have numbers for enum names.

Reply:
Those names are decent, I agree they could be better, but I think this works.